models/model.js

```diff
@@ -14,26 +14,34 @@ exports.fetchSpecArt = (iD)=>{
     })
 }
 
-exports.fetchAllArt = ()=>{
-    let artQuery = 
-    `SELECT 
-    articles.article_id,
-    articles.author,
-    articles.title,
-    articles.topic,
-    articles.created_at,
-    articles.votes,
-    articles.article_img_url,
-    COUNT(comments.article_id) AS comment_count
-    FROM articles
-    LEFT JOIN comments 
-    ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC`
-
+exports.fetchAllArt = ()=>{ 
+    return db.query(
+        `SELECT 
+        articles.article_id,
+        articles.author,
+        articles.title,
+        articles.topic,
+        articles.created_at,
+        articles.votes,
+        articles.article_img_url,
+        COUNT(comments.article_id) AS comment_count
+        FROM articles
+        LEFT JOIN comments 
+        ON articles.article_id = comments.article_id
+        GROUP BY articles.article_id
+        ORDER BY articles.created_at DESC`)
+    .then(({rows})=>{
+            return rows
+        })
+}
 
-    return artPromise = db.query(artQuery).then(({rows})=>{
+exports.fetchAllComs = (iD)=>{
+    let query = `SELECT * FROM comments` 
+    query += ` WHERE article_id = ${iD}`
+    query += ` ORDER BY created_at`
+    return db.query(query).then(({rows})=>{
+        if(rows.length < 1) return Promise.reject({status:404 ,msg:'article does not exist'})
```

Comment:
Here I would make another call to the database before rejecting the promise to see if the article id exists. If it does but there's no rows, this should be a 200 because it's a valid request, so you would expect an empty comments array to be returned to the controller. If it doesn't exist, then it's a 404 reject